sensehat/src/main/java/com/google/android/things/contrib/driver/sensehat/SenseHat.java

```diff
@@ -16,19 +17,149 @@
 
 package com.google.android.things.contrib.driver.sensehat;
 
+import android.graphics.Color;
+
+import com.google.android.things.contrib.driver.hts221.Hts221;
+import com.google.android.things.contrib.driver.hts221.Hts221SensorDriver;
+import com.google.android.things.contrib.driver.lps25h.Lps25h;
+import com.google.android.things.contrib.driver.lps25h.Lps25hSensorDriver;
+import com.google.android.things.pio.I2cDevice;
+import com.google.android.things.pio.PeripheralManagerService;
+
 import java.io.IOException;
 
 /**
- * Driver factory for the Sense Hat.
+ * Driver factory for Raspberry Pi Sense HAT.
  */
 @SuppressWarnings({"unused", "WeakerAccess"})
-public class SenseHat {
-    public static final int I2C_ADDRESS =  0x46;
-    public static final String BUS_DISPLAY = "I2C1";
+public class SenseHat implements AutoCloseable {
+
     public static final int DISPLAY_WIDTH = LedMatrix.WIDTH;
     public static final int DISPLAY_HEIGHT = LedMatrix.HEIGHT;
 
-    public static LedMatrix openDisplay() throws IOException {
-        return new LedMatrix(BUS_DISPLAY);
+    private static final String I2C_BUS = "I2C1";
+    private static final int I2C_ADDRESS = 0x46;
+
+    private static final String JOYSTICK_INTERRUPT = "BCM23"; // Interrupt pin for joystick events
+
+    private static final int SENSE_HAT_REG_WHO_AM_I = 0xF0;
+
+    protected static I2cDevice i2cDevice;
```

Comment:
instead of the `static` maybe we could have a `ATTiny` driver with 2x inner class `Joystick` and `Display`? what do you think?

I'm happy to send a pull request against your fork if you agree we should iterate in that direction.